File: app/src/main/java/top/someones/cardmatch/ui/workshop/ModAdapter.java
package top.someones.cardmatch.ui.workshop;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import top.someones.cardmatch.databinding.ModInfoLayoutBinding;
import top.someones.cardmatch.entity.Mod;;

public class ModAdapter extends RecyclerView.Adapter<ModAdapter.ViewHolder> {

    private final List<Mod> mModList;
    private final ModOnClickListener mOnClickListener;

    public ModAdapter(List<Mod> modList, ModOnClickListener modOnClickListener) {
        this.mModList = modList;
        this.mOnClickListener = modOnClickListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(ModInfoLayoutBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Mod mod = mModList.get(position);
        if (mOnClickListener != null) {
            holder.binding.getRoot().setOnClickListener(l -> {
                mOnClickListener.onClick(mod);
            });
        }
        holder.binding.modCover.setImageBitmap(mod.getCover());
        holder.binding.modName.setText(mod.getName());
        holder.binding.modAuthor.setText("作者：".concat(mod.getAuthor()));
        holder.binding.modVersion.setText("版本：".concat(String.valueOf(mod.getVersion())));
    }

    @Override
    public int getItemCount() {
        return mModList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        final ModInfoLayoutBinding binding;

        public ViewHolder(ModInfoLayoutBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    public interface ModOnClickListener {
        void onClick(Mod mod);
    }
}

File: app/src/main/java/top/someones/cardmatch/ui/workshop/ModAdapter.kt
package top.someones.cardmatch.ui.workshop

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import top.someones.cardmatch.databinding.ModInfoLayoutBinding
import top.someones.cardmatch.entity.WorkShopMod
import top.someones.cardmatch.util.NetUtil

class ModAdapter(
    private val mModList: List<WorkShopMod>,
    private val onClick: ((it: WorkShopMod) -> Unit)?
) : RecyclerView.Adapter<ModAdapter.ViewHolder>() {

    private lateinit var context: Context

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        context = parent.context
        return ViewHolder(
            ModInfoLayoutBinding.inflate(
                LayoutInflater.from(context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val mod = mModList[position]
        onClick?.let {
            holder.binding.root.setOnClickListener { _: View? -> it.invoke(mod) }
        }
        Glide.with(context)
            .load("${NetUtil.MAIN}mod/${mod.uuid}/img")
            .into(holder.binding.modCover)
        holder.binding.modName.text = mod.name
        holder.binding.modAuthor.text = "作者：${mod.author}"
        holder.binding.modVersion.text = "版本：${mod.version}"
    }

    override fun getItemCount(): Int {
        return mModList.size
    }

    class ViewHolder(val binding: ModInfoLayoutBinding) : RecyclerView.ViewHolder(
        binding.root
    )
}